### PR TITLE
excelCompare: init at 0.6.1

### PR DESCRIPTION
**Pull request by ilikeavocadoes**
###### Motivation for this change

Add utility to produce text diff of Excel files. Couldn't build from source since it requires Java version <= 6 which is not in Nixpkgs.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/text/excelCompare/default.nix
Comment (symphorien) [inline pkgs/tools/text/excelCompare/default.nix]: you don't need to rename excel_cmp before wrapping: `wrapProgram` is an equivalent of `makeWrapper` which wraps in place.
But reading the script it would be better to use the approach of https://nixos.org/nixpkgs/manual/#sec-language-java
that is:
* put all jars in $out/share/excel-cmp
* wrap java exactly as they do in the example, setting the class path to $out/share/excel-cmp/*

File: pkgs/tools/text/excelCompare/default.nix
Comment (symphorien) [inline pkgs/tools/text/excelCompare/default.nix]: makewrapper and unzip should be nativeBuildInputs

**Comment by symphorien** [comment]
This binary distribution redistributes a number of third party libraries: are they all redistributable under the MIT licence ?

File: pkgs/tools/text/excelCompare/default.nix
Comment (ilikeavocadoes) [reply pkgs/tools/text/excelCompare/default.nix]: Thanks for the review. In the example for makeWrapper, what am I supposed to put in place of `org.foo.Main`? I tried `com.ka.spreadsheet.diff.SpreadSheetDiffer` as is in the original `excel_cmp` startup script, but it results in `Error: Could not find or load main class .nix.store.3ykvmyv0ihvdpvxvdcivld3jf89blj83-ExcelCompare-0.6.1.share.excel_cmp.commons-digester-1.8.jar`.

**Comment by ilikeavocadoes** [comment]
> This binary distribution redistributes a number of third party libraries: are they all redistributable under the MIT licence ?

I will try and find the licenses of the third party libraries. In case they have a different license, what should I put as the `meta.license`?

File: pkgs/tools/text/excelCompare/default.nix
Comment (symphorien) [reply pkgs/tools/text/excelCompare/default.nix]: this works for me:
```diff
diff --git a/pkgs/tools/text/excelCompare/default.nix b/pkgs/tools/text/excelCompare/default.nix
index a0ec897eb5e..b433aa5e518 100644
--- a/pkgs/tools/text/excelCompare/default.nix
+++ b/pkgs/tools/text/excelCompare/default.nix
@@ -1,4 +1,4 @@
-{stdenv, lib, fetchurl, makeWrapper, unzip, jdk}:
+{stdenv, lib, fetchurl, makeWrapper, unzip, jre}:
 
 stdenv.mkDerivation rec {
   name = "ExcelCompare-${version}";
@@ -7,16 +7,14 @@ stdenv.mkDerivation rec {
     url = "https://github.com/na-ka-na/ExcelCompare/releases/download/${version}/ExcelCompare-${version}.zip";
     sha256 = "1304qraazqm8mgjgnijzl070l0haa8jds7spnsy3xwh3vda0ka4x";
   };
-  buildInputs = [ jdk unzip makeWrapper ];
+  nativeBuildInputs = [ unzip makeWrapper ];
   sourceRoot = ".";
 
   installPhase = ''
-    mkdir -p $out
-    cp -r bin $out/bin
-    chmod 555 $out/bin/excel_cmp
-    mv $out/bin/excel_cmp $out/bin/excel_cmp.sh
-    makeWrapper $out/bin/excel_cmp.sh $out/bin/excel_cmp \
-      --prefix PATH : ${lib.makeBinPath [ jdk ]}
+    install -Dt $out/share/excelcmp bin/dist/*
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/excelcmp \
+      --add-flags "-ea -cp $out/share/excelcmp/\* com.ka.spreadsheet.diff.SpreadSheetDiffer"
   '';
 
   meta = {
```
the trick is to prevent the wildcard `*` from being expanded by the shell

**Comment by symphorien** [comment]
meta.license can be a list, but I don't know if it means license A OR B or A AND B.

**Comment by symphorien** [comment]
Also, please fix the ofborg failure, apparently meta.maintainers does not evaluate.

**Comment by ilikeavocadoes** [comment]
> Also, please fix the ofborg failure, apparently meta.maintainers does not evaluate.

Seems like I hadn't already added myself to maintainer list, while I thought I had. I made a pull request of it (https://github.com/NixOS/nixpkgs/pull/55512). Should I wait until it gets merged and then rebase on master?

**Comment by ilikeavocadoes** [comment]
I couldn't make sense of the 3rd party licenses, nor how they should be (or should they be) included in the meta attributes, so I just removed that field for the time being.

**Comment by ilikeavocadoes** [comment]
The source repo has a directory of the third party licenses: https://github.com/na-ka-na/ExcelCompare/tree/master/legal

Now the issue seems to be that how can this be included in meta? Also, I'm absolutely clueless about these licenses' implications to the redistributability of this software, as my understanding about the copyright laws is zero.

**Comment by symphorien** [comment]
you can write something like
```nix
{
meta.license = with stdenv.lib.licenses; [ mit asl20 ];
}
```
I have no idea whether it means OR or AND.

**Comment by aanderse** [comment]
@ilikeavocados I would suggest filling an issue upstream to get clarification on the licensing situation so we can proceed with this.

**Review by ryantm — CHANGES_REQUESTED** [review]
Please fix evaluation errors.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity. Feel free to reopen the discussion.

**Pull request closed without merging**
